src/api/app/models/bs_request.rb

```diff
@@ -78,6 +78,8 @@ class SaveError < APIError
   scope :find_for, ->(params) { BsRequest::FindFor::Query.new(params).all }
   scope :obsolete, -> { where(state: OBSOLETE_STATES) }
 
+  scope :by_actions_target_projects, ->(project_ids) { joins(:bs_request_actions).where(bs_request_actions: { target_project_id: project_ids }) }
```

Comment:
Why is this now a scope here?

I mean, instead of using the BsRequest class to "filter" the records that have a BsRequestAction that targets the project. I feel more "natural" to use the project and ask for it requests that have an action pointing to it.

The responsibility must be on the side of the project, not on the side the BsRequest class.

Reply:
@mdeniz:

> Why is this now a scope here?
>

The scope was created here after @hennevogel 's [review](#discussion_r225578534). For me creating the scope in the BsRequest model makes sense because what is being returned is a list of BsRequests. If it was a list of projects it would be fine for me to create a scope in the project model.

> I mean, instead of using the BsRequest class to "filter" the records that have a BsRequestAction that targets the project. It feels more "natural" to use the project and ask for it requests that have an action pointing to it.
>
> The responsibility must be on the side of the project, not on the side the BsRequest class.

This sounds to me more like implementation details in the scope.

I'm fine with implementing one or the other option. Can we live with this approach for the moment, or is it really required to change to the project method approach you proposed before?

Reply:
> It feels more "natural" to use

for me it sounds like personal taste. I'm fine with scope on ```BsRequest``` too..

Reply:
Is NOT an scope... that's the point. You don't need to scope the requests, you just get the requests related to the project (from the project that is who knows which one are).

This is all about who knows what... and you shouldn't ask to the BsRequest class to know which of them are related to a single project through the BsRequestAction... Instead you must ask the project about that. The project is who knows how the requests are related to it.

Reply:
When I say "more natural" is about this:

How you get the requests using your solution:
```ruby
BsRequest.all.by_actions_target_projects(project.id)
```

and this is how you get it if you have it in the project side (better name for the method is needed):
```ruby
project.stageable_requests
```

Reply:
BTW, looking at the relationship by using the scope **by_actions_target_projects** is looking at this from a too low level of abstraction.

I mean, what you want to know in your code is how to get the stageable request of a project, you shouldn't have to know how this relationship is implemented (right now using BsRequestAction to target the project). And if this implementation changes one day you must change your scope naming and where it used. But if you keep it in the project side, you only need to change the content of that method/relationship and not the code that uses it. 

So, this is also about having code that doesn't suffer with changes.

Reply:
I agree with @mdeniz, I think is more natural having this in the project side. IMHO, having it in a scope adds unnecessary complexity (more difficult to understand, at least).

Reply:
I also agree with @mdeniz. You can also use a scoped assocation:
https://api.rubyonrails.org/classes/ActiveRecord/Associations/ClassMethods.html#method-i-has_many

Reply:
Ok. Changed back `scope :by_actions_target_projects` to method `stageable_requests` in Project model. Please re-review.